isis/tests/FunctionalTestsLronaccal.cpp

```diff
@@ -0,0 +1,173 @@
+#include <QTemporaryDir>
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+#include "Endian.h"
+#include "PixelType.h"
+#include "lronaccal.h"
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "Histogram.h"
+#include "crop.h"
+
+using namespace std;
+using namespace Isis;
+using namespace testing; 
+
+/**
+  *
+  * @brief Calibration application for the LRO NAC cameras
+  *
+  *  lronaccal performs radiometric corrections to images acquired by the Narrow Angle
+  *  Camera aboard the Lunar Reconnaissance Orbiter spacecraft.
+  *
+  * @author 2016-09-16 Victor Silva
+  *
+  * @internal
+  *   @history 2022-04-26 Victor Silva - Original Version - Functional test is against known value for input 
+  *                                      cub since fx is not yet callable
+  */
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/lronaccal.xml").expanded();
+
+TEST(LronaccalDefault, FunctionalTestsLronaccal) {
+  QTemporaryDir outputDir;
+
+  ASSERT_TRUE(outputDir.isValid());
+
+  /*This application can not be run on any image that has been
+    geometrically transformed (i.e. scaled, rotated, sheared, or
+    reflected) or cropped
+  */
+  QString iCubeFile =  "data/lronaccal/input/M1333276014R.cub";
+  QString oCubeFile = outputDir.path() + "/out.default.cub";
+  QString oCubeCropFile = outputDir.path() + "/out.default.crop.cub";
+  QString tCubeFile = "data/lronaccal/truth/M1333276014R.default.crop.cub";
```

Comment:
What is tcube doing here? it's not used for output but used to check histograms against anyways? Even it was output it should be going into a temp directory.

Reply:
It is comparing against truth cubes so no they aren't output cubes. But again, using as opposed to hard-coded values. But there wasn't guidance as to what would be preferred. I guess I thought it would be better to switch out truth cubes as changes to app occur and the code would not need to be changed in tests.